src/Configuration/Option/Prefix.php

```diff
@@ -28,7 +29,7 @@ class Prefix implements OptionInterface
      */
     public function filter(OptionsInterface $options, $value)
     {
-        if (is_callable($value)) {
+        if (is_callable($value) && (is_object($value) || $value instanceof Closure)) {
```

Comment:
@vladvildanov Can we get your eyes on this?

The `is_callable()` check has been around for 7+ years and the issue is that if the prefix is set to `"chr"` or `"abs"` the PHP functions would be called.

I can't think of an elegant solution to prevent this, without creating edge cases.

Reply:
@tillkruss I think here it's enough to keep `$value instanceof Closure`, because that's exactly what we want to ensure. We have two different flows for string and Closure, so the only issue is that `is_callable()` consider strings matching php default functions as callable.

Reply:
Okay, are you certain that only checking `$value instanceof Closure` won't break peoples installations?

Reply:
I am, but test coverage should prove this better

Reply:
The `object` + `Closure` check makes sense, but I'm uncomfortable merging this because we have one bug report of this causing an issue and an unknown number of people using:

```php
$client = new Predis\Client($server, [
    'prefix' => 'my_prefix_callback',
    'prefix' => 'MyPrefix::callback',
]);
```

Reply:
@tillkruss Okay, lets leave `is_callable()` call for now, so people could see deprecation message and have time to fix it in their code. We can substitute all `call_user_func($callable)` with `$callable()` later on.

For now, we can add a hack like this `if (is_callable($prefix) && !function_exists($prefix))`, so prefixes corresponding to php function name won't consider as callables.

Reply:
That seems reasonable, even tho it's a breaking change. @mmodler what do you think?

Reply:
I think this would solve the issue in my case.